[CI] PutSynonymsActionRequestSerializingTests testEqualsAndHashcode failing

**Build Scans:**
- [elasticsearch-periodic #11245 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/gjxqfyzmwmwt2)
- [elasticsearch-periodic-platform-support #11225 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/llh5jb3va3q6o)
- [elasticsearch-periodic-platform-support #11218 / almalinux-8-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/agdl7grehwvdw)
- [elasticsearch-periodic-platform-support #11218 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/tjwif4uip2w54)
- [elasticsearch-periodic-platform-support #11218 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/6w2icqpe4khoc)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.action.synonyms.PutSynonymsActionRequestSerializingTests.testEqualsAndHashcode" -Dtests.seed=F87B0FACAE033B30 -Dtests.locale=es-MX -Dtests.timezone=America/North_Dakota/Center -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.action.synonyms.PutSynonymsActionRequestSerializingTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testEqualsAndHashcode'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Request mutation should not be equal to original
Expected: not <org.elasticsearch.action.synonyms.PutSynonymsAction$Request/unset>
     but: was <org.elasticsearch.action.synonyms.PutSynonymsAction$Request/unset>
```

**Issue Reasons:**
- [main] 5 failures in test testEqualsAndHashcode (2.1% fail rate in 239 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/test/java/org/elasticsearch/action/synonyms/PutSynonymsActionRequestSerializingTests.java
+++ b/server/src/test/java/org/elasticsearch/action/synonyms/PutSynonymsActionRequestSerializingTests.java
@@ -14,6 +14,8 @@ import org.elasticsearch.synonyms.SynonymRule;
 import org.elasticsearch.test.AbstractWireSerializingTestCase;
 
 import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
 
 import static org.elasticsearch.action.synonyms.SynonymsTestUtils.randomSynonymsSet;
 
@@ -32,14 +34,14 @@ public class PutSynonymsActionRequestSerializingTests extends AbstractWireSerial
     @Override
     protected PutSynonymsAction.Request mutateInstance(PutSynonymsAction.Request instance) throws IOException {
         String synonymsSetId = instance.synonymsSetId();
-        SynonymRule[] synonymRules = instance.synonymRules();
+        List<SynonymRule> synonymRules = Arrays.asList(instance.synonymRules());
         boolean refresh = instance.refresh();
         switch (between(0, 2)) {
             case 0 -> synonymsSetId = randomValueOtherThan(synonymsSetId, () -> randomIdentifier());
-            case 1 -> synonymRules = randomValueOtherThan(synonymRules, () -> randomSynonymsSet());
+            case 1 -> synonymRules = randomValueOtherThan(synonymRules, () -> Arrays.asList(randomSynonymsSet()));
             case 2 -> refresh = refresh == false;
             default -> throw new AssertionError("Illegal randomisation branch");
         }
-        return new PutSynonymsAction.Request(synonymsSetId, synonymRules, refresh);
+        return new PutSynonymsAction.Request(synonymsSetId, synonymRules.toArray(new SynonymRule[0]), refresh);
     }
 }

